presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimate.java

```diff
@@ -83,15 +82,7 @@ public PlanNodeStatsEstimate mapOutputRowCount(Function<Double, Double> mappingF
     public PlanNodeStatsEstimate mapSymbolColumnStatistics(Symbol symbol, Function<SymbolStatsEstimate, SymbolStatsEstimate> mappingFunction)
     {
         return buildFrom(this)
-                .setSymbolStatistics(symbolStatistics.entrySet().stream()
-                        .collect(Collectors.toMap(
-                                Map.Entry::getKey,
-                                e -> {
-                                    if (e.getKey().equals(symbol)) {
-                                        return mappingFunction.apply(e.getValue());
-                                    }
-                                    return e.getValue();
-                                })))
+                .addSymbolStatistics(symbol, mappingFunction.apply(symbolStatistics.get(symbol)))
```

Comment:
That is the core of this improvement: original map is not copied. Instead only one column is added to existing, immutable map. Pcollections PMap guarantees that such operation is performant